src/oscar/apps/partner/strategy.py

```diff
@@ -204,7 +204,7 @@ class StockRequired(object):
     def availability_policy(self, product, stockrecord):
         if not stockrecord:
             return availability.Unavailable()
-        if not product.get_product_class().track_stock:
+        if not product.get_product_class().track_stock and stockrecord.price_excl_tax:
```

Comment:
This seems to make products which are free (price is 0) unavailable. Not sure we want, perhaps:

    if not stockrecord or stockrecord.price_excl_tax is None:
        return availability.Unavailable()